web3/providers/websocket.py

```diff
@@ -80,25 +63,19 @@ def __init__(self, endpoint_uri=None, websocket_kwargs=None):
                     '{0} are not allowed in websocket_kwargs, '
                     'found: {1}'.format(RESTRICTED_WEBSOCKET_KWARGS, found_restricted_keys)
                 )
-        self.conn = PersistentWebSocket(
-            self.endpoint_uri, WebsocketProvider._loop, websocket_kwargs
+        WebsocketProvider.conn = WebSocketConnection(
```

Comment:
Any reason why `conn` was made a class variable? Every `WebsocketProvider` should have its own connection object, so `self` seems to make more sense to me.